st2common/st2common/services/triggers.py

```diff
@@ -117,17 +107,17 @@ def _get_trigger_api_given_rule(rule):
                             (trigger_ref.pack, triggertype_ref.pack))
     else:
         trigger_dict['parameters'] = rule.trigger.get('parameters', {})
+        if not trigger_dict['parameters']:
```

Comment:
This should fix one of the things we've talked about yesterday, right?

Anyway, it seems like we could still simply a code a bit.